multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -138,4 +154,55 @@ private FileMetadata parseFileEntry(FileEntry fileEntry) {
                                     .namespace(fileEntry.getNamespace())
                                     .build();
     }
+
+    private FileEntry uploadFileFromUrl(String spaceGuid, String namespace, String mtarEndpoint) throws FileStorageException, IOException {
+        if (!UriUtil.isValidUrl(mtarEndpoint)) {
+            throw new SLException("Invalid URL");
+        }
+        String decodedUrl = URLDecoder.decode(mtarEndpoint, StandardCharsets.UTF_8);
+        WebClient client = new RestUtil().createWebClient(false);
+
+        ClientResponse response = client.get()
+                                        .uri(decodedUrl)
+                                        .exchange()
+                                        .block();
+        long fileSize = response.headers()
+                                .contentLength()
+                                .orElseThrow(() -> new SLException("MTAR endpoint did not return Content-Length header"));
+        long maxUploadSize = new Configuration().getMaxUploadSize();
+        if (fileSize > maxUploadSize) {
```

Comment:
Is this the only validation for downloading a bigger MTAR than allowed? From a security PoV, the server could return a Content-Length header with a small value, but proceed to send an enormous MTAR.

Reply:
Added a `LimitedInputStream` wrapper to the content stream